collated/main/A0121261Y.md

```diff
@@ -0,0 +1,856 @@
+     *
+     * @param userInput full user input string
+     * @return the command based on the user input
+     */
+    public Command parseCommand(String userInput) {
+        final Matcher matcher = BASIC_COMMAND_FORMAT.matcher(userInput.trim());
+        if (!matcher.matches()) {
+            return new IncorrectCommand(String.format(MESSAGE_INVALID_COMMAND_FORMAT, HelpCommand.MESSAGE_USAGE));
+        }
+
+        final String commandWord = matcher.group("commandWord").toLowerCase();
+        final String arguments = matcher.group("arguments");
+        switch (commandWord) {
+
+        case AddCommand.COMMAND_WORD:
+            return prepareAdd(arguments);
+
+        case SelectCommand.COMMAND_WORD:
+            return prepareSelect(arguments);
+
+        case DeleteCommand.COMMAND_WORD:
+            return prepareDelete(arguments);
+
+        case ClearCommand.COMMAND_WORD:
+            return new ClearCommand();
+
+        case FindCommand.COMMAND_WORD:
+            return prepareFind(arguments);
+
+        case ListCommand.COMMAND_WORD:
+            return prepareList(arguments);
+
+        case ExitCommand.COMMAND_WORD:
+            return new ExitCommand();
+
+        case HelpCommand.COMMAND_WORD:
+            return new HelpCommand();
+
+        case ViewCommand.COMMAND_WORD:
+        	return prepareView(arguments);
+
+        case UndoCommand.COMMAND_WORD:
+            return new UndoCommand();
+
+        case RedoCommand.COMMAND_WORD:
+        	return new RedoCommand();
+
+        case EditCommand.COMMAND_WORD:
+            return prepareEdit(arguments);
+
+        case DoneCommand.COMMAND_WORD:
+            return prepareMark(arguments);
+
+        case ConfirmCommand.COMMAND_WORD:
+            return new ConfirmCommand();
+
+        default:
+            return new IncorrectCommand(MESSAGE_UNKNOWN_COMMAND);
+        }
+    }
+
+    /**
+     * Parses arguments in the context of the add mark command.
+     *
+     * @param arguments
+     * @return
+     */
+    private Command prepareMark(String args) {
+        Optional<Integer> index = parseIndex(args);
+        if(!index.isPresent()){
+            return new IncorrectCommand(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, DoneCommand.MESSAGE_USAGE));
+        }
+
+        return new DoneCommand(index.get());
+    }
+
+```
+###### \java\seedu\address\logic\parser\Parser.java
+``` java
+    /**
+     * Parses arguments in the context of the add task command.
+     *
+     * case 0: no date and time , i.e no args for time found
+     * case 1: date only
+     * case 2: date and startTime
+     * case 3: date and startTime endTime
+     *
+     * @param args full command args string
+     * @return the prepared command
+     */
+    private Command prepareAdd(String args){
+        String[] dateTimeArgs = null;
+        TaskType taskType;
+ //       final Matcher matcher = TASK_DATA_ARGS_FORMAT.matcher(args.trim());
```

Comment:
Don't leave commented out code. Just go ahead and delete unused code. Now that we are using a revision control system, we can recover any deleted code later.